collector/collector.go

```diff
@@ -0,0 +1,29 @@
+package collector
+
+import (
+	"log/slog"
+
+	"github.com/LambdaLabs/redfish_exporter/config"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stmcginnis/gofish"
+)
+
+// Collector creates and returns a prometheus.Collector from the Module, based on the Prober.
+// Both redfish client and logger are common dependencies for any redfish_exporter collector,
+// and must be provided as inputs.
+func NewCollectorFromModule(m *config.Module, rfClient *gofish.APIClient, logger *slog.Logger) prometheus.Collector {
+	switch m.Prober {
+	case "gpu_collector":
+		return NewGPUCollector(rfClient, logger, &m.GPUCollector)
```

Comment:
I'd rather not to use a pointer for a Config to copy the data, especially for a small struct not to accidentally share the config and avoid a bug behavior, but in this case, it doesn't matter much as a config is frequently reloaded and collector is recreated every time a request comes